### PR TITLE
Document Stacks

### PR DESCRIPTION
**Pull request by SpComb**
* Update the quickstart example to use a simple wordpress+mysql stack

    Should we use `kontena stack pull` to fetch the stackfile, or directly `kontena stack install examples/wordpress`?

    Getting the private IP address for vagrant is rather difficult ATM, but the `WORDPRESS_DB_HOST=%{project}-mysql.kontena.local` part is nicely simplified, and the auto-generated vault secrets are :+1: 

* Brief architecture section

* Stacks page

 * Document most `kontena stack ...` commands
 * Document some of the stack DNS namespace details
 * Mention the stack expose mechanism

## TODO

* Document remaining `kontena stack ...` commands with examples
  * `kontena stack build`
  * `kontena stack upgrade`
* Document the YAML stack file fields in full - probably copy-pasta from the existing `kontena app` -> `kontena.yml` reference?
* Stacks extends
* Stack load balancer linking
* Iter-stack linking?
* Publishing different services from multiple stacks via a single `strategy: daemon` kontena-lb service

### CODE REVIEW

**Comment by jnummelin** [comment]
SOmething for the todo list: How exposing services from stack works.

File: docs/getting-started/quick-start.md [hunk 1]
```diff
@@ -78,24 +78,23 @@ If you followed the steps above, you should now have a working Kontena setup ins
...
 
- Now you are ready to deploy your first application. In this section we will show you how to initiate a simple WordPress application and deploy it to your Kontena Grid.
+ Now you are ready to deploy your first application stack.
+ In this section we will show you how to package a simple WordPress application and deploy it to your Kontena Grid.
 
 First create the `kontena.yml` file with the following contents:
 
 ```
-version: '2'
+stack: examples/wordpress
```
Comment (jakolehm) [inline docs/getting-started/quick-start.md]: We could use more "advanced" example here... like:

```
stack: examples/wordpress
version: 1.0.0
variables:
  db_password:
    type: string
    from:
      vault: WORDPRESS_DB_PASSWORD
      random_string: 20
    to:
      vault: WORDPRESS_DB_PASSWORD
services:
  wordpress:
    image: wordpress:4-apache
    stateful: true
    ports:
      - 80:80
    secrets:
      - secret: WORDPRESS_DB_PASSWORD
        name: WORDPRESS_DB_PASSWORD
        type: env

    environment:
      - WORDPRESS_DB_HOST=mysql
      - WORDPRESS_DB_USER=root
  mysql:
    image: mariadb:5.5
    stateful: true
    secrets:
      - secret: WORDPRESS_DB_PASSWORD
        name: MYSQL_ROOT_PASSWORD
        type: env
```

File: docs/getting-started/quick-start.md [hunk 2]
```diff
@@ -105,21 +104,114 @@ services:
...
-$ kontena app deploy
+You can then install and deploy the `wordpress` stack:
+
+```
+$ kontena stack install --deploy kontena.yml
+ [done] Creating stack wordpress      
+ [done] Deploying stack wordpress     
+```
+
+The intitial stack deployment may take some time while the host nodes pull the necessary Docker images.
```
Comment (jakolehm) [inline docs/getting-started/quick-start.md]: intitial -> initial

File: docs/getting-started/quick-start.md [hunk 3]
```diff
@@ -105,21 +104,114 @@ services:
...
+    wordpress:
+      image: wordpress:4.1
+      status: running
+      revision: 1
+      stateful: yes
+      scaling: 1
+      strategy: ha
+      deploy_opts:
+        min_health: 0.8
+      dns: wordpress.wordpress.development.kontena.local
```
Comment (jakolehm) [inline docs/getting-started/quick-start.md]: wordpress.wordpress.development.kontena.local -> wordpress.wordpress.test.kontena.local

File: docs/getting-started/quick-start.md [hunk 4]
```diff
@@ -105,21 +104,114 @@ services:
...
+    mysql:
+      image: mariadb:5.5
+      status: running
+      revision: 1
+      stateful: yes
+      scaling: 1
+      strategy: ha
+      deploy_opts:
+        min_health: 0.8
+      dns: mysql.wordpress.development.kontena.local
```
Comment (jakolehm) [inline docs/getting-started/quick-start.md]: wordpress.wordpress.development.kontena.local -> wordpress.wordpress.test.kontena.local

File: docs/getting-started/quick-start.md [hunk 5]
```diff
@@ -105,21 +104,114 @@ services:
...
+      ip: 10.81.128.30
+      public ip: 192.0.2.1
+      status: running
+      exit code: 0
+```
+
+You use the public IP address and published port numbers associated with any service container to access the service. **Note:** For the special case of using Vagrant for the Kontena setup, you must use the *private* IP address of the node running the `wordpress/wordpress` service:
+
+```
+$ kontena node show core-01
```
Comment (jakolehm) [inline docs/getting-started/quick-start.md]: Use some generated name here, like `dry-pond-86`

File: docs/getting-started/quick-start.md [hunk 6]
```diff
@@ -105,21 +104,114 @@ services:
...
+core-01:
+  id: XI4K:NPOL:EQJ4:S4V7:EN3B:DHC5:KZJD:F3U2:PCAN:46EV:IO4A:63S5
+  agent version: 1.0.0.pre1
```
Comment (jakolehm) [inline docs/getting-started/quick-start.md]: 1.0.0.pre1 -> 1.0.0

**Comment by SpComb** [comment]
Updated quickstart with https://stacks.kontena.io/stack/terom/wordpress/version/0.3.0, using stack vault variables. NB: they are shared across any instance of the wordpress stack.

Still need to re-run with a plugin-deployed 1.0.0.pre for more authentic examples (grid=test etc).

File: docs/using-kontena/stacks.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+Multiple instances of a stack file can be installed on the same Kontena grid, assigning an unique name to each deployed stack.
+The service containers for each stack are run within a separate `$stack.$grid.kontena.local` DNS namespace.
+The service containers can connect to the bare DNS names of other services in the same stack.
+Stacks can also expose a service, allowing other stacks to use the bare DNS name of the exposing stack to connect to the exposed service.
+
+## Usage
+
+Kontena Stacks can be distributed as YAML files via the ***Stack Registry***, deployed via the ***Kontena Master*** to run as ***Service Containers*** within the ***Grid***.
+
+### Stack Registry
```
Comment (jakolehm) [inline docs/using-kontena/stacks.md]: Check latest updates from stack registry pr... these commands have been moved.

File: docs/using-kontena/stacks.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+terom/wordpress
+```
+
+Omit the search term to list all available stacks.
+
+#### `kontena stack info terom/wordpress`
+
+Lists available versions for a given stack.
+
+```
```
Comment (jakolehm) [inline docs/using-kontena/stacks.md]: Output has been improved, needs update.

**Comment by SpComb** [comment]
What do we do with our [docs/references/kontena-yml.md](https://github.com/kontena/kontena/blob/master/docs/references/kontena-yml.md)?

The services keys are AFAIK largely the same, but many other details differ from the `kontena app` format.

**Comment by jnummelin** [comment]
LGTM
There's always room for improvement but this is a good starting point

**Pull request merged**